cluster-autoscaler/proposals/provisioning-request.md

```diff
@@ -0,0 +1,285 @@
+# Provisioning Request CRD
+
+author: kisieland
+
+## Background
+
+Currently CA does not provide any way to express that a group of pods would like
+to have a capacity available.  
+This is caused by the fact that each CA loop picks a group of unschedulable pods
+and works on provisioning capacity for them, meaning that the grouping is random
+(as it depends on the kube-scheduler and CA loop interactions).  
+This is especially problematic in couple of cases:
+
+  - Users would like to have all-or-nothing semantics for their workloads.
+    Currently CA will try to provision this capacity and if it is partially
+    successful it will leave it in cluster until user removes the workload.
+  - Users would like to lower e2e scale-up latency for huge scale-ups (100
+    nodes+). Due to CA nature and kube-scheduler throughput, CA will create
+    partial scale-ups, e.g. `0->200->400->600` rather than one `0->600`. This
```

Comment:
All or nothing/Batched workloads are an interesting concept, woudl the same be provided for scale up or is it explicitly for scale down?

Reply:
THe current proposal is to just provide the atomicity at scale-up path and let the scale-down take nodes as those become unneeded.